pkg/docker/adapter.go

```diff
@@ -276,8 +275,12 @@ func (d *DockerContainerAdapter) ApplyConfig(proposedConfig *agent.ProposedConfi
 			return err
 		}
 
-		// We make the assumption every setting is a number parsed as float
-		err = pg.AlterSystem(d.PGDriver, knobConfig.Name, strconv.FormatFloat(knobConfig.Setting.(float64), 'f', -1, 64))
+		// Get the setting value using the proper type conversion method
+		settingValue, err := knobConfig.GetSettingValue()
+		if err != nil {
+			return fmt.Errorf("failed to get setting value for %s: %w", knobConfig.Name, err)
+		}
+		err = pg.AlterSystem(d.PGDriver, knobConfig.Name, settingValue)
```

Comment:
Just realised we set paramaters one by one. We should actually try parse all setting values first before applying, to prevent an issue where we only partially apply a config